plugins/woocommerce/includes/wc-product-functions.php

```diff
@@ -1834,3 +1835,34 @@ function wc_product_attach_featured_image( $attachment_id, $product = null, $sav
 	}
 }
 add_action( 'add_attachment', 'wc_product_attach_featured_image' );
+
+/**
+ * Delete related product transients when a product is saved.
```

Comment:
Reasoning behind this nuclear approach https://github.com/woocommerce/woocommerce/issues/30349#issuecomment-2683037010